### DISCUSSION TITLE
Support compose.override.yml files

### DISCUSSION THREAD

**Discussion by 17Halbe** [louislam/dockge#415] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API

### 🔖 Feature description

Support display an editability of compose.override.yml files 

### ✔️ Solution

Add a second compose.yml box below the current one to display a compose.override.yml. See [Docker Docs: Compose.override.yml](https://docs.docker.com/compose/multiple-compose-files/merge/)
Could probably accomplished by accordion styled "cards" dropdown of both files.
Bonus: Add a button next to the compose.yml box to create an override file.

### ❓ Alternatives

Option 1: 
One could just output the compose compiled file and render the box uneditable/disabled. See: [compose config](https://docs.docker.com/engine/reference/commandline/compose_config/)
Option 2:
Display the merged file (see Option 1) and make only the override file editable, since this is the only file which should be edited in production anyway.
Option 3: 
Display the override file instead of the compose file.

### 📝 Additional Context

It is becoming more and more common to ship compose.yml files in a package and overriding user specific needs in an override file. 
I'm using this approach myself; but you may also take a look into mailcow, simfony, caddy-proxy etc.
Btw this might also be an option for Dockge, to distribute it's compose.yml file and only add an override for user specific settings. ;)
I am interested in providing a PR on this if you'd like..

**Comment by lohrbini** [comment]
Maybe this could be done by adding override files in `util.ts` and create new editbox in `compose.vue`

```
diff --git a/frontend/src/pages/Compose.vue b/frontend/src/pages/Compose.vue
index 7d37854..180d116 100644
--- a/frontend/src/pages/Compose.vue
index 587e6dd..000a509 100644
--- a/common/util-common.ts
+++ b/common/util-common.ts
@@ -120,9 +120,11 @@ export const allowedRawKeys = [
 
 export const acceptedComposeFileNames = [
     "compose.yaml",
+    "compose.yml",
     "docker-compose.yaml",
     "docker-compose.yml",
-    "compose.yml",
+    "docker-compose.override.yml",
+    "docker-compose.override.yaml"
 ];
 
 /**
diff --git a/frontend/src/pages/Compose.vue b/frontend/src/pages/Compose.vue
index 7d37854..180d116 100644
--- a/frontend/src/pages/Compose.vue
+++ b/frontend/src/pages/Compose.vue
@@ -182,6 +182,24 @@
                         {{ yamlError }}
                     </div>
 
+                    <!-- YAML editor docker-compose.override.yml -->
+                    <div class="shadow-box mb-3 editor-box" :class="{'edit-mode' : isEditMode}">
+                        <h4 class="mb-3">override config</h4>
+                        <prism-editor
+                            ref="editor"
+                            v-model="stack.composeYAML"
+                            class="yaml-editor"
+                            :highlight="highlighterYAML"
+                            line-numbers :readonly="!isEditMode"
+                            @input="yamlCodeChange"
+                            @focus="editorFocus = true"
+                            @blur="editorFocus = false"
+                        ></prism-editor>
+                    </div>
+                    <div v-if="isEditMode" class="mb-3">
+                        {{ yamlError }}
+                    </div>
+
                     <!-- ENV editor -->
                     <div v-if="isEditMode">
                         <h4 class="mb-3">.env</h4>
```

Draft result (probably not working correct):
![image](https://github.com/louislam/dockge/assets/10088089/4d981c3a-94d7-48c7-b02b-bba4c655bc93)


If someone wants to work on this the changes are committed here: https://github.com/lohrbini/dockge/tree/feat/dockge-override-support

This is just a "quick draft" can have a look at it after current PR

**Comment by eliasfloreteng** [comment]
I also implemented this feature in #872 and you can use the image from my fork if you change `louislam/dockge:1` to `ghcr.io/eliasfloreteng/dockge:compose-override-editor` in your docker compose.
